python/doc/examples/calibration/least_squares_and_gaussian_calibration/plot_calibration_withoutobservedinputs.py

```diff
@@ -0,0 +1,175 @@
+import openturns.viewer as viewer
+
+# %%
+# The vector of parameters is 
+# :math:`\boldsymbol{\theta} = (a, b, c)^T \in \mathbb{R}^3`. 
+# This model is linear in :math:`\boldsymbol{\theta}` and identifiable.
+# It is derived from the equation:
+# 
+# .. math::
+#     y(x) = a + b x + c x^2
+# 
+# at :math:`x=-1.0, -0.6, -0.2, 0.2, 0.6, 1.0`. 
+# In the model, however, the abscissas are fixed and constant. 
+# Therefore, the parametric model has 3 parameters, no input and 6 outputs 
+# :math:`y_1, ..., y_6`.
+# This produces a set of 5 observations for each output, leading to a total 
+# of 5 (observations per output) * 6 (outputs) = 30 observations.
+g = ot.SymbolicFunction(
+    ["a", "b", "c"],
+    [
+        "a +  -1.0  * b +  1.0  * c",
+        "a +  -0.6  * b +  0.36  * c",
+        "a +  -0.2  * b +  0.04  * c",
+        "a +  0.2  * b +  0.04  * c",
+        "a +  0.6  * b +  0.36  * c",
+        "a +  1.0  * b +  1.0  * c",
+    ],
+)
+outputDimension = g.getOutputDimension()
+print(outputDimension)
+
+# %%
+# We set the true value of the parameters.
+trueParameter = ot.Point([12.0, 7.0, -8.0])
+print(trueParameter)
+
+# %%
+# In order to generate the observed outputs, we create a distribution 
+# in dimension 3, with Dirac (i.e. constant) marginals.
+Theta1 = ot.Dirac(trueParameter[0])
+Theta2 = ot.Dirac(trueParameter[1])
+Theta3 = ot.Dirac(trueParameter[2])
+inputRandomVector = ot.ComposedDistribution([Theta1, Theta2, Theta3])
+
+# %%
+# The candidate value is chosen to be different to the true parameter value. 
+candidate = ot.Point([8.0, 9.0, -6.0])
+calibratedIndices = [0, 1, 2]
+model = ot.ParametricFunction(g, calibratedIndices, candidate)
+
+# %%
+# We consider a multivariate gaussian noise with zero mean, standard deviation 
+# equal to 0.05 and independent copula.
+# The independent copula implies that the errors of the 6 outputs are independent.
+outputObservationNoiseSigma = 1.0
+meanNoise = ot.Point(outputDimension)
+covarianceNoise = ot.Point(outputDimension, outputObservationNoiseSigma)
+R = ot.IdentityMatrix(outputDimension)
+observationOutputNoise = ot.Normal(meanNoise, covarianceNoise, R)
+print(observationOutputNoise)
+
+# %%
+# Finally, we generate the outputs by evaluating the exact outputs of the 
+# function and adding the noise.
+# We use a sample with size equal to 5.
+size = 5
+# Generate exact outputs
+inputSample = inputRandomVector.getSample(size)
+outputStress = g(inputSample)
+# Add noise
+sampleNoise = observationOutputNoise.getSample(size)
+outputObservations = outputStress + sampleNoise
+
+# %%
+# Now is the important part of this script : there is no input observations. 
+# This is why we create a sample with size equal to 5 and dimension equal to 0. 
+# Even if the calibration model has no inputs observations, observed inputs 
+# are required by the current programming interface and this is why we have to 
+# create this object. 
+inputObservations = ot.Sample(size, 0)  # Trick
```

Comment:
I Calibration classes should allow for an `inputObservations` argument set via `Sample()` : ie no need to set the size, any 0-dim sample should be considered as empty. If the class wants to set the size to simplify the computations it can do so internally.

Reply:
I agree that this is not satisfactory. Your suggestion, however, is not related to this example, but suggests a (relevant) change in the API. So I suggest to do this in a separate PR.